src/object/cli_obj.c

```diff
@@ -1718,27 +1718,23 @@ uint32_t
 dc_obj_retry_delay(tse_task_t *task, int err, uint16_t *retry_cnt, uint16_t *inprogress_cnt,
 		   uint32_t timeout_sec)
 {
-	uint32_t	delay = 0;
-	uint32_t	limit = 4;
+	uint32_t delay = 0;
+
+	if (err == -DER_INPROGRESS || err == -DER_UPDATE_AGAIN)
+		++(*inprogress_cnt);
 
 	/*
-	 * Randomly delay 5 ~ 1028 us if it is not the first retry for
-	 * -DER_INPROGRESS || -DER_UPDATE_AGAIN cases.
+	 * Randomly delay 5 ~ 1028 us if it is not the first retry.
 	 */
-	++(*retry_cnt);
-	if (err == -DER_INPROGRESS || err == -DER_UPDATE_AGAIN) {
-		if (++(*inprogress_cnt) > 1) {
-			limit += *inprogress_cnt;
-			if (limit > 10)
-				limit = 10;
+	if (++(*retry_cnt) > 1) {
```

Comment:
One question.  If there is a pool map refresh, do we still do this delay?

Reply:
Hmm, I thought the mechanism for overload is controlled by the server side. Not sure if we should delay here for all cases. 

@liw @wangshilong  any concern here?

Reply:
For dtx retry case, should we retry immediately or delay? @Nasf-Fan

Reply:
> Hmm, I thought the mechanism for overload is controlled by the server side. Not sure if we should delay here for all cases.
> 
> @liw @wangshilong any concern here?

DER_OVERLOAD_RETRY has its own delay assignment in this function, i think it ok.